zeroex/orderwatch/order_watcher_test.go

```diff
@@ -956,17 +1019,18 @@ func TestOrderWatcherBatchEmitsAddedEvents(t *testing.T) {
 	orderEventsChan := make(chan []*zeroex.OrderEvent, 10)
 	orderWatcher.Subscribe(orderEventsChan)
 
+	// TODO(albrow): Create orders in a single batch when scenario supports it.
 	signedOrders := []*zeroex.SignedOrder{}
 	for i := 0; i < 2; i++ {
-		signedOrder := scenario.CreateZRXForWETHSignedTestOrder(t, ethClient, makerAddress, takerAddress, big.NewInt(1000), big.NewInt(1000))
-		// Creating a valid order involves transferring sufficient funds to the maker, and setting their allowance for
-		// the maker asset. These transactions must be mined and Mesh's BlockWatcher poller must process these blocks
-		// in order for the order validation run at order submission to occur at a block number equal or higher then
-		// the one where these state changes were included. With the BlockWatcher poller configured to run every 200ms,
-		// we wait 500ms here to give it ample time to run before submitting the above order to the Mesh node.
-		time.Sleep(500 * time.Millisecond)
+		signedOrder := scenario.NewSignedTestOrder(t, orderopts.SetupMakerState(true))
 		signedOrders = append(signedOrders, signedOrder)
 	}
+	// Creating a valid order involves transferring sufficient funds to the maker, and setting their allowance for
```

Comment:
@jalextowle I moved this sleep statement. I think we don't need to sleep after _each_ order is created, just at the end. Can you confirm this is okay?

Reply:
Nice catch. My only concern would be that we aren't giving the blockwatcher enough time to catch up to potentially two blocks. My sense is that it probably is enough time, but do you have a sense on this?

Reply:
I think the amount of time we need to sleep scales with the number of events we need to process, not the number of blocks. It should be proportional to `numberOfEvents / 500` rounded down. So in this case, I don't think we will need to increase the time. I say we leave it as is and adjust it if it turns out that we aren't waiting long enough.